### ISSUE TITLE
Cannon hangs checking deployment versions while running offline on v2.4.21

### ISSUE THREAD

**Issue by FuzzB0t** [usecannon/cannon#315]
this function hangs forever. Needs to fail gracefully and just carry on

```
async function checkCannonVersion(currentVersion) {
    const latestVersion = await execPromise('npm view @usecannon/cli version');
// ... cut
```
 
then even though all packages are built locally and running things offline before had no problems, this error now pops up
```
> cannon run dex:0.0.0
Starting local node...

settings not configured: please create file ~/.local/share/cannon/settings.json for better performance. See https://usecannon.com/docs for more information.
using default settings (https://ipfs.io, https://ethereum.publicnode.com/)
Error: deployment not found: dex:0.0.0. please make sure it exists for the main preset and network 13370
```

and also this occurs trying to build
`Error: deployment not found: mintable-token:latest. please make sure it exists for preset main and network 13370.`

**Comment by dbeal-eth** [comment]
In general for this issue we want a complete cannon build to be able to occur in the absence of an internet connection as long as required package artifacts had already been downloaded

This may also involve auto-caching builds

**Comment by FuzzB0t** [comment]
closed with #323
